scipy/stats/_survival.py

```diff
@@ -374,3 +391,138 @@ def _ecdf_right_censored(sample):
+    ...     right=[31, 47, 80, 82, 82, 149]
+    ... )
+    >>> y = stats.CensoredData(
+    ...     uncensored=[10, 10, 12, 13, 14, 15, 16, 17, 18, 20, 24, 24,
+    ...                 25, 28,30, 33, 35, 37, 40, 40, 46, 48, 76, 81,
+    ...                 82, 91, 112, 181],
+    ...     right=[34, 40, 70]
+    ... )
+
+    We can calculate and visualize the empirical survival functions
+    of both groups as follows.
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> ax = plt.subplot()
+    >>> ecdf_x = stats.ecdf(x)
+    >>> ax.step(np.insert(ecdf_x.x, 0, 0),
+    ...         np.insert(ecdf_x.sf.points, 0, 1),
+    ...         where='post', label='Astrocytoma')
+    >>> ecdf_y = stats.ecdf(y)
+    >>> ax.step(np.insert(ecdf_y.x, 0, 0),
+    ...         np.insert(ecdf_y.sf.points, 0, 1),
+    ...         ls='-.', where='post', label='Glioblastoma')
+    >>> ax.set_xlabel('Time to death (weeks)')
+    >>> ax.set_ylabel('Empirical SF')
+    >>> plt.legend()
+    >>> plt.show()
+
+    Based on a visual inspection of the survival functions, we see a difference
+    but cannot conclude about its significance.
+
+    Next, we will use the logrank test to assess wether the difference
+    between the two survival functions is significant.
+
+    >>> res = stats.log_rank(x=x, y=y)
+    >>> res.statistic
+    7.49659416854
+    >>> res.pvalue
+    0.006181578637
+
+    Using a significance level of 5%, we would reject the null hypothesis in
+    favor of the alternative hypothesis: "there is a difference between the
+    two survival functions".
+
+    """
+    x = _iv_CensoredData(sample=x, param_name='x')
+    y = _iv_CensoredData(sample=y, param_name='y')
+
+    xy = CensoredData(
+        uncensored=np.concatenate((x._uncensored, y._uncensored)),
+        right=np.concatenate((x._right, y._right))
+    )
+
+    res = ecdf(xy)
+    idx = res.sf._d.astype(bool)  # events were observed (not censored)
+    times_xy = res.x[idx]
+    at_risk_xy = res.sf._n[idx]
+    deaths_xy = res.sf._d[idx]
+
+    n_died_x = x._uncensored.size
+
+    res_x = ecdf(x)
+    i = np.searchsorted(res_x.x, times_xy)  # or use `interpolate_1d`
+    at_risk_x = np.append(res_x.sf._n, 0)[i]  # 0 at risk after last x time
+    at_risk_y = at_risk_xy - at_risk_x
+
+    sum_exp_deaths_x = np.sum(at_risk_x * (deaths_xy/at_risk_xy))
+
+    # equivalent: formulation with the mean
+    # statistic = (
+    #     (n_died_x - sum_exp_deaths_x)**2/sum_exp_deaths_x
+    #     + (n_died_y - sum_exp_deaths_y)**2/sum_exp_deaths_y
+    # )
+    # warning occurs when multiple deaths at same time
+    with np.errstate(invalid='ignore'):
+        sum_var = np.nansum(
+            at_risk_x*at_risk_y*deaths_xy*(at_risk_xy - deaths_xy)
+            / (at_risk_xy**2*(at_risk_xy - 1))
+        )
+    statistic = (n_died_x - sum_exp_deaths_x)**2/sum_var
+
+    pvalue = norm.sf(np.sqrt(statistic))*2
+    # equivalent to chi2(df=1).sf(statistic) but allow alternative
```

Comment:
I think we need to decide whether we're going to offer one-sided alternatives before we choose this as the statistic. 

If we want one-sided alternatives, I think we should return the signed, unsquared version of this statistic. (Or at least include the signed, unsquared version in another attribute.) I understand that references often give the squared version, but squaring it loses information about the sign, which is important when doing a one-sided test. This also complicates the addition of a permutation version of this test, which may be important for small samples.

Reply:
To start with I just matched the output of R but I agree that this is wrong. Here we return the statistic for the chi2 and not the normal distribution. I would suggest not squaring, as you suggest, and in the doc describe that we use the normal distribution and that the statistic corresponds to that and not chi2. Hence a difference (of a square) if someone would compare both.

Reply:
I'm really not sure because as you say, other software returns the square. I'm torn between agreement with other software and utility.

I would be ok with returning the squared statistic as `statistic` and the un-squared one as `zstatistic` or something.

Reply:
Squaring often reduces the amount of information. But you are the the experts here.

Reply:
Yes, that is why I raised this concern. I was just torn because I wasn't sure about breaking the precedent set by other software. Looks like this has converged on returning the unsquared result, so I'll resolve this. With proper documentation, I think it's OK.